src/driver-roccat-kone-aimo.c

```diff
@@ -0,0 +1,1220 @@
+	uint8_t sensitivity;                        // 0x06 means 0 because its -5 to 5 in the UI, so 1-11
+	uint8_t x_y_linked;                         // Set X and Y dpi separately, but its not a feature in ROCCAT Swarm, default `1F`
+	uint8_t current_dpi;                        // 0-4 for the 5 resolutions
+	uint16_t xres[ROCCAT_NUM_DPI];              // 5 resolutions saved to switch between. value * 50 = DPI
+	uint16_t yres[ROCCAT_NUM_DPI];              // 5 resolutions saved to switch between. value * 50 = DPI
+	uint8_t report_rate;                        // 0 = 125 hz, 1 = 250hz, 2 = 500hz, 3 = 1000hz
+	uint8_t angle_snapping;                     // 0 = off, 1 = on
+	struct roccat_color unk_color1;             // 08 FF 07 00 not sure, looks like some kind of 0 intensity blue. Nothing in the official software changes it.
+	uint8_t lighting_effect;                    // From 0x01 to 0x04 : fixed, blinking, breathing, beating
+	uint8_t lighting_effect_speed;              // From 0x01 to 0x03
+	struct roccat_color brightness;             // intensity is global brightness, color is unused but `1D 13 FF` default.
+	struct roccat_color unk_color2;             // FF 59 FF 00, looks like some kind of default gradient, might be AIMO gradient?
+	struct roccat_color unk_color3;             // FF FD FD 00
+	struct roccat_color unk_color4;             // FF F4 64 00
+	struct roccat_color unk_color5;             // FF F4 00 00
+	uint8_t unknown1;                           // FF default, doesn't change by any UI settings.
+	struct roccat_color led_scrollwheel_color;  // only solid
+	struct roccat_color led_leftstrip_color_1;  // 4 color gradient. Solid is just all same color
+	struct roccat_color led_leftstrip_color_2;
+	struct roccat_color led_leftstrip_color_3;
+	struct roccat_color led_leftstrip_color_4;
+	struct roccat_color led_rightstrip_color_1; // 4 color gradient. Solid is just all same color
+	struct roccat_color led_rightstrip_color_2;
+	struct roccat_color led_rightstrip_color_3;
+	struct roccat_color led_rightstrip_color_4;
+	struct roccat_color led_leftblob_color;    // only solid
+	struct roccat_color led_rightblob_color;   // only solid
+	uint8_t custom_or_theme;                   // 00-09 for custom based on theme, theme is 80-89
+	uint8_t unknown2;                          // 01 default, doesn't change by any UI settings.
+	uint8_t padding[6];
+	uint16_t checksum;
+} __attribute__((packed));
+_Static_assert(sizeof(struct roccat_settings_report) == ROCCAT_REPORT_SIZE_SETTINGS, "Size of roccat_buttons is wrong");
+
+struct roccat_macro {
+	uint8_t report_id;                           // 0x08
+	uint8_t bank;                               // 0x01 or 0x02
+	uint8_t profile;
+	uint8_t button_index;
+	uint8_t repeats;                            // Number of repetition for this macro
+	char group[ROCCAT_MACRO_GROUP_NAME_LENGTH]; // Folder name
+	char name[ROCCAT_MACRO_NAME_LENGTH];
+	uint16_t length;
```

Comment:
Note: there is no `length` field in your `macro_hex_dump1.pat`.

Reply:
It's supposed to be there. I might've forgotten to save the pattern after fixing it. Without it, all the data is shifted incorrectly.